### DISCUSSION TITLE
Better adjustment and alignment of core usage widgets

### DISCUSSION THREAD

**Discussion by MadhavJivrajani** [pesos/grofer#95] [General]
`grofer` will most likely not work on large cored machines. Currently, the widgets for the per-core usage are only adjusted in height and have constant width (relative to the terminal dimensions). 

It might be better to have some way to adjust these bars in columns as well depending on the number of cores available on the underlying machine. For example: `htop` flawlessly shows core usage on a 96 core machine, whereas `grofer` ended up crashing. We'll probably need to investigate the reason for this crash a little more, but it's definitely a good idea to get started on better aligning and placing the widgets on the screen

**Comment by anihm136** [comment]
Would it make sense to have a different visualization for large cored machines? I don't think 96 bars are very informative

**Reply by MadhavJivrajani** [reply]
Bars are an effective visualisation, but just not the way `grofer` does it, for example here's what `htop` looks like on a 96 core machine 

![Screenshot from 2021-02-02 17-06-08.png](https://user-images.githubusercontent.com/12381034/106621021-24b1d280-6598-11eb-9180-22d497d629d4.png)

**Reply by MadhavJivrajani** [reply]
What other forms of virtualization could be  possible?

**Reply by anihm136** [reply]
I was thinking just a percentage with a color. But yeah, `htop` looks fine for 96 cores

**Reply by anihm136** [reply]
How feasible would a circular graph be?

**Reply by anirudhRowjee** [reply]
perhaps a rectangular matrix (say 8 *12) of percentage values with colors? it'll look something like a heatmap
